nistats/tests/test_verify_events_file_uses_tab_separators.py

```diff
@@ -93,23 +98,33 @@ def test_binary_opening_an_image():
     img_data = bytearray(
             b'GIF87a\x01\x00\x01\x00\xe7*\x00\x00\x00\x00\x01\x01\x01\x02\x02'
             b'\x07\x08\x08\x08\t\t\t\n\n\n\x0b\x0b\x0b\x0c\x0c\x0c\r;')
-    with NamedTemporaryFile(mode='wb', suffix='.gif',
-                            dir=os.getcwd()) as temp_img_obj:
-        temp_img_obj.write(img_data)
-        with assert_raises(ValueError):
-            _verify_events_file_uses_tab_separators(
```

Comment:
So looking more at the code, I think the problem is more related to how `_verify_events_file_uses_tab_separators` works: internally it reopens the same file which is forbidden on Windows.

In the test, the `NamedTemporaryFile` returns a file object (already opened) that you try to reopen in `_verify_events_file_uses_tab_separators` and here windows is unhappy. Your proposed solution suffers the same problem but it's worse because you hide it using a try/except.

I would suggest to refactor `_verify_events_file_uses_tab_separators` to accept file object parameters and when used not reopen it. I think this will avoid lots of changes proposed by this PR.

Reply:
I have looked at the code and here are some thoughts:

> I think this will avoid lots of changes proposed by this PR.

There were ~13 tests failing on Windows and this accounts for only 3-4 of them. The other changes I have made are not because of this function.

> I would suggest to refactor _verify_events_file_uses_tab_separators to accept file object parameters ...

It maybe that the function could have used better design principles but this was not caught when it was being written.
Refactoring it now would also mean some code modification everywhere the function is being used, and that would & should be a separate PR, not this one. I recognize that this code fragility is symptomatic of bad code design, and I will address it immediately, just not in this PR.

It seems to me backwards to rewrite the testee function to make the tests work with it, instead of modifying the tests to work with the function being tested. The only immediate benefit is that we will continue using the `with` statement instead of manually doing a `try-except-finally`, which is what `with` essentially does.

I propose that we keep these changes here so we have testing on Windows platform for subsequent PRs which will be forthcoming soon & open a new PR to redesign this function & associated tests.